### PR TITLE
feat use zig cc to target glibc 2.17

### PR DESCRIPTION
**Pull request by lewis6991**
- Does not work with `Debug` or `RelWithDebInfo`

- [x] Test it actually works
   - https://github.com/neovim/neovim-releases/actions/runs/9820554858/job/27115268509

### CODE REVIEW

**Comment by clason** [comment]
If we're building on the host, we can probably avoid the cmake hackery and just use the version ubuntu-latest comes with. @dundargoc 

(The "prepare container" step can also be removed.)

**Comment by lewis6991** [comment]
> (The "prepare container" step can also be removed.)

Completely? Don't we still need to install those packages even on host?

**Comment by clason** [comment]
You're already installing most of them later; just add the missing one (libfuse) there?

File: .github/workflows/release.yml
Comment (clason) [inline .github/workflows/release.yml]:
```suggestion
    runs-on: ubuntu-latest
```
(24.04 is still in beta)

Comment (lewis6991) [reply .github/workflows/release.yml]: 24.04 comes with cmake 3.28 (our min version). 23.10 comes with cmake 3.27

So the choice is use a beta runner, or cmake install hackery.

Comment (clason) [reply .github/workflows/release.yml]: Minimum version is 3.13 tho. (Not sure why this workflow specified 3.28; I assume it was just "if we have to install manually, might at least use the latest".)

Comment (lewis6991) [reply .github/workflows/release.yml]: ah ok. The MINIMUM_CMAKE_VERSION var lead me to believe it was 3.28

**Comment by lewis6991** [comment]
Release is smaller now https://github.com/neovim/neovim-releases/releases/tag/nightly-test

EDIT: forget to restore RelWithDebInfo. Now it's the same.

File: .github/workflows/release.yml
Comment (dundargoc) [inline .github/workflows/release.yml]: Probably not needed.

File: .github/workflows/release.yml
Comment (dundargoc) [inline .github/workflows/release.yml]: You can remove this after merging the PR on neovim repo

**Review by dundargoc — APPROVED** [review]
Looks good, nice work.

Don't forget to update the README to reflect this

File: .github/workflows/release.yml
Comment (lewis6991) [reply .github/workflows/release.yml]: We'll need it to respin 0.10

Comment (dundargoc) [reply .github/workflows/release.yml]: Huh, fair. Backport the neovim pr?

File: .github/workflows/release.yml
Comment (clason) [inline .github/workflows/release.yml]:
```suggestion
      # Not needed for 0.10.1+ : https://github.com/neovim/neovim/pull/29599
```
Or `TODO(lewis): remove after 0.10.1 release`

**Comment by lewis6991** [comment]
All releases updated successfully. Merging.

**Comment by dundargoc** [comment]
Nice work! This will unblock a few other improvements as well.

**Pull request merged**
